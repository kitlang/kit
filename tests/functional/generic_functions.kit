function genericFunction[T](fmt: CString, v: T) {
    printf(fmt, v);
}

function genericFunction2[T, U](fmt: CString, t: T, u: U) {
    printf(fmt, t, u);
}

function main() {
    genericFunction("%i\n", 1);
    genericFunction("%i\n", 2);
    genericFunction("%.2f\n", 3.0);
    genericFunction("%s\n", "hello");

    genericFunction2("%s%i\n", "hello", 1);
    genericFunction2("%i%s\n", 1, "hello");
}
