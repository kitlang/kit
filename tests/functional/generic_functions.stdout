1
2
3.00
hello
hello1
1hello
